cmake/KokkosConfigCommon.cmake.in

```diff
@@ -209,6 +209,290 @@ int main()
   set(${_VAR} ${_RET} PARENT_SCOPE)
 endfunction()
 
+# this function searches a given target recursively looking for transitive dependency to Kokkos
+#
+#       FOUND_DEPENDENCY    --> output: is set to ON if a dependency to kokkos is found
+#       LIBRARY_KEYWORD     --> input: LINK_LIBRARIES or INTERFACE_LINK_LIBRARIES (keyword used with get_target_properties)
+#       DEPENDENT_LIBRARY   --> input: The library to search for in the target properties
+#       INVESTIGATED_TARGET --> input: target to be checked for transitive dependency
+#
+function(kokkos_check_dependency FOUND_DEPENDENCY LIBRARY_KEYWORD DEPENDENT_LIBRARY INVESTIGATED_TARGET)
+  if(TARGET ${INVESTIGATED_TARGET})
+    get_target_property(target_dependencies ${INVESTIGATED_TARGET} ${LIBRARY_KEYWORD})
+    if(target_dependencies)
+      foreach(dependency IN LISTS target_dependencies)
+        kokkos_check_dependency(depends_transitively ${LIBRARY_KEYWORD} ${DEPENDENT_LIBRARY} ${dependency})
+        if(depends_transitively)
+          set(transitive_dependency ON)
+        endif()
+      endforeach()
+    endif()
+
+    if(${DEPENDENT_LIBRARY} IN_LIST target_dependencies OR transitive_dependency)
+      set(${FOUND_DEPENDENCY} ON PARENT_SCOPE)
+    endif()
```

Comment:
Wouldn't it make sense to check for `if(${DEPENDENT_LIBRARY} IN_LIST target_dependencies)` first and only search for `transitive_dependency` if there is no direct dependency?

Reply:
yes, would save some traversal.